c/bpxskt.c

```diff
@@ -369,18 +369,39 @@ Socket *tcpClient3(SocketAddress *socketAddress,
       /* manual says returnCode and value only meaningful if return value = -1 */
       *returnCode = 0;
       *reasonCode = 0;
+      if (readTimeoutInMillis >= 0) {
+        setSocketOption(socket, SOL_SOCKET, SOCK_SO_RCVTIMEO, sizeof(int), (char*)&readTimeoutInMillis, returnCode, reasonCode);
+        setSocketBlockingMode(socket, TRUE, returnCode, reasonCode);
```

Comment:
You may also want to set a non-zero send time-out because this blocking mode may lead to that zss thread hanging forever, waiting for the peer socket to become available for writes.